tests/lib/rules/no-implicit-coercion.js

```diff
@@ -248,6 +248,17 @@ ruleTester.run("no-implicit-coercion", rule, {
                 type: "BinaryExpression"
             }]
         },
+        {
+            code: "`${foo}`",
+            output: "String(foo)",
+            options: [{ templateString: true }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{
+                messageId: "useRecommendation",
+                data: { recommendation: "String(foo)" },
+                type: "TemplateLiteral"
+            }]
+        },
```

Comment:
Can you add more tests to cover all the checks for the new option:

* valid tests when the option's value is `true`:
  * `` `a${foo}` ``
  * `` `${foo}b` ``  
  * `` `${foo}{bar}` ``  
  * `` tag`${foo}` ``
* valid tests for `` `${foo}` ``:
  * when there are no options specified
  * when the options are an empty object `{}`
  * when the option is explicitly set to `false`

Reply:
I missed some options and had to go afk. I'll add the last 3 tests later today.